src/compoenents/CustomerMailDetailPannel/IotDetailPannel.tsx

```diff
@@ -1,77 +1,88 @@
-    { date: '2024-07-05', value: 26.05 },
-    { date: '2024-07-06', value: 21.05 },
-    { date: '2024-07-07', value: 27.05 },
-    { date: '2024-07-08', value: 20.05 },
-    { date: '2024-07-09', value: 25.05 },
-    { date: '2024-07-10', value: 20.05 },
-    { date: '2024-07-11', value: 21.05 },
-    { date: '2024-07-12', value: 23.45 },
-    { date: '2024-07-13', value: 20.05 },
-    { date: '2024-07-14', value: 23.45 },
-    { date: '2024-07-15', value: 27.05 },
-    { date: '2024-07-16', value: 21.05 },
-    { date: '2024-07-17', value: 28.05 },
-    { date: '2024-07-18', value: 22.05 },
-    { date: '2024-07-19', value: 25.05 },
-    { date: '2024-07-20', value: 24.05 },
-    { date: '2024-07-21', value: 21.05 },
-    
-  ];
-
-  const humidityData = [
-    { date: '2024-07-01', value: 65 },
-    { date: '2024-07-02', value: 67 },
-    { date: '2024-07-03', value: 70 },
-    // More data points...
-  ];
-
-  const soilMoistureData = [
-    { date: '2024-07-01', value: 45 },
-    { date: '2024-07-02', value: 47 },
-    { date: '2024-07-03', value: 50 },
-    // More data points...
-  ];
-
-  const uvLevelData = [
-    { date: '2024-07-01', value: 5 },
-    { date: '2024-07-02', value: 6 },
-    { date: '2024-07-03', value: 7 },
-    // More data points...
-  ];
-
-  const altitudeData = [
-    { date: '2024-07-01', value: 1000 },
-    { date: '2024-07-02', value: 1020 },
-    { date: '2024-07-03', value: 1015 },
-    // More data points...
-  ];
-
-  const atmosphericPressureData = [
-    { date: '2024-07-01', value: 1012 },
-    { date: '2024-07-02', value: 1010 },
-    { date: '2024-07-03', value: 1013 },
-    // More data points...
-  ];
+  const [iotData, setIotData] = useState<IoTData[]>([]);
+
+  const API_BASE_URL = import.meta.env.VITE_API_BASE_URL;
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await fetch(`${API_BASE_URL}/iot`);
+        const result = await response.json();
+
+        console.log("Fetched API Data:", result);
+        setIotData(result);
+      } catch (error) {
+        console.error("Error fetching IoT data:", error);
+      }
+    };
+
+    fetchData();
+  }, [API_BASE_URL]);
+
+  // Generic function to filter data based on the field name (e.g., temperature, humidity, etc.)
+  const getFilteredData = (key: keyof IoTData, intervalMinutes: number = 60) => {
+    const filteredData = iotData
+      .filter(item => item[key] !== undefined) // Check if the key exists
+      .map(item => ({
+        date: `${item.recordedDate}T${item.recordedTime}`, // Combine date and time
+        value: item[key] as number // Use the key to access the value dynamically
+      }));
+
+    // Group the filtered data by the specified time interval (e.g., every 60 minutes)
+    const groupedData = groupDataByTime(filteredData, intervalMinutes);
+
+    console.log(`Grouped data for ${key}:`, groupedData); // Log the grouped data for debugging
```

Comment:
When we merge with dev branch keep in mind we need to remove these debugging console logs from code base